tests/unit_server/rest_handler_test.py

```diff
@@ -271,13 +283,41 @@ async def fn2(*args, **Kwargs):
 
     handler.get_authenticated_user = MagicMock(side_effect=fn2)
 
-    request.body = '{"code": "thecode", "state": "state"}'
+    request.body = '{"code": "thecode", "state": "state"}'  # invalid state
```

Comment:
Can you explain this more?  What's invalid about this state?  It won't pass `_decode_state()`, but that's mocked out.  Technically it's legal according to the oauth2 spec.

Reply:
I missed that mocking. I figured that `_decode_state()` was raising the `HTTPError` and triggered `handler.write()`

Reply:
it looks that `get_authenticated_user()` raises the error, yeah?

Reply:
That should also be mocked out?

Is the test diff just pulling each case apart into separate functions, and there's no actual test changes?

Reply:
ah I see now https://github.com/WIPACrepo/rest-tools/blob/argparse-args/rest_tools/server/handler.py#L460